### PR TITLE
added lhn-backbutton

### PR DESCRIPTION
**Pull request by Taiwrash**
### Details

- Fixes the LHN button to redirect to the right slugs when the `back button` is clicked. I updated the `navigateBack` function.

### Fixed Issues

PROPOSAL:
$ https://github.com/Expensify/App/issues/42600
$ https://github.com/Expensify/App/issues/42600#issuecomment-2175586994

### Tests

- Click the back button and it works
- Try the two slugs currently available and it works
- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

https://github.com/Expensify/App/assets/49725691/6f71100f-dcf0-4bf8-ba9d-911b14e160a7

### CODE REVIEW

**Comment by dubielzyk-expensify** [comment]
I've been OOO so might have missed something but this recording seems wrong to me. If you navigate to:
`Expensify Classic` > `Chat` > `Edit or delete messages` then click **Back**, to me it shouldn't take me back to `Expensify Classic`. 

Instead I'd expect back to show the previous page and LHN which in this instance would be `Chat`. 

Unless I'm very confused. cc @Expensify/design

**Comment by rushatgabhane** [comment]
@dubielzyk-expensify yeah that's right it should take you to the hub

**Comment by rushatgabhane** [comment]
@Taiwrash please fix it

**Comment by shawnborton** [comment]
Totally agree @dubielzyk-expensify - I feel like I had similar comments about this back button behavior elsewhere, I can't remember where though.

**Comment by shawnborton** [comment]
Ah yes, @Taiwrash please take another look at the comments I had on [your closed PR here](https://github.com/Expensify/App/pull/43910#issuecomment-2186080416).

**Comment by Taiwrash** [comment]
Oh! I can't actually remember where exactly I got the suggestion to make it the way I did before. 

### Below is the update made
- When you go to `Expensify Classic > Chat > Edit or delete messages` then click Back, it will show the previous page and LHN which in this instance would be `Chat` 
> See below
https://github.com/Expensify/App/assets/49725691/50c9808f-b2e0-47d8-bd2e-e34b0cf3d185

**Comment by shawnborton** [comment]
I think that seems better, thanks!

**Comment by Taiwrash** [comment]
Hello everyone, I am still awaiting our feedback to merge this @rushatgabhane @dubielzyk-expensify @shawnborton

**Comment by dubielzyk-expensify** [comment]
That looks good to me too. Can you confirm that the same back behaviour also happens with the back button on the browser?

**Comment by shawnborton** [comment]
That looks and feels pretty good to me.

File: .gitconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[commit]
+    gpgsign = true
+[user]
+    email = rasheedrtm1@gmail.com
+    name = Rasheed Mudasiru
+    signingkey = 30C8087C891F6D07
+[gpg]
```
Comment (rushatgabhane) [inline .gitconfig:7]: @Taiwrash please remove this commit

File: docs/Gemfile.lock [hunk 2]
```diff
@@ -259,6 +261,7 @@ PLATFORMS
   arm64-darwin-23
   x86_64-darwin-20
   x86_64-darwin-21
+  x86_64-linux
```
Comment (rushatgabhane) [inline docs/Gemfile.lock:264]: same with this file. please remove this commit

File: docs/assets/js/main.js [hunk 3]
```diff
@@ -49,27 +49,41 @@ function isInRange(num, min, max) {
  * If a page is directly accessed (e.g., via deep link, bookmark, or opened in a new tab),the user will be navigated
  * back to the relevant hub page of that article.
  */
+let newPath;
```
Comment (rushatgabhane) [inline docs/assets/js/main.js:52]: Why is this variable on in the global scope? Can we move it to function

File: docs/assets/js/main.js [hunk 4]
```diff
@@ -49,27 +49,41 @@ function isInRange(num, min, max) {
...
-    setTimeout(toggleHeaderMenu, 250);
+    if (newPath !== currentPath) {
+        window.location.href = newPath;
+        
+        // Add a little delay to avoid showing the previous content in a fraction of a time
+        setTimeout(toggleHeaderMenu, 250);
+    }
 }
 
+// Add an event listener for the popstate event
```
Comment (rushatgabhane) [inline docs/assets/js/main.js:81]: This comment is not needed, code is self explainatory

File: docs/assets/js/main.js [hunk 5]
```diff
@@ -309,4 +325,4 @@ window.addEventListener('DOMContentLoaded', () => {
         // Count property of y-axis to keep scroll position & reference it later for making the body fixed when sidebar opened
         document.documentElement.style.setProperty('y-axis', `${window.scrollY}px`);
     });
-});
+});
```
Comment (rushatgabhane) [inline docs/assets/js/main.js:328]: please add a new line to make github happy

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

**Comment by Taiwrash** [comment]
@rushatgabhane all corrections made

**Comment by Taiwrash** [comment]
Looking forward to the next review @rushatgabhane @shawnborton @dubielzyk-expensify @r8

File: docs/Gemfile.lock [hunk 6]
```diff
@@ -215,6 +215,8 @@ GEM
       racc (~> 1.4)
     nokogiri (1.15.4-x86_64-darwin)
       racc (~> 1.4)
+    nokogiri (1.15.4-x86_64-linux)
```
Comment (rushatgabhane) [inline docs/Gemfile.lock:218]: Please remove these changes as they're not required

File: docs/assets/js/main.js [hunk 7]
```diff
@@ -309,4 +327,5 @@ window.addEventListener('DOMContentLoaded', () => {
         // Count property of y-axis to keep scroll position & reference it later for making the body fixed when sidebar opened
         document.documentElement.style.setProperty('y-axis', `${window.scrollY}px`);
     });
-});
+    
+});
```
Comment (rushatgabhane) [inline docs/assets/js/main.js:331]: add a new line

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

**Comment by Taiwrash** [comment]
@rushatgabhane just fixed all

**Comment by Taiwrash** [comment]
Still await the final review for this, I want to pick another issue to work on

**Comment by Taiwrash** [comment]
@rushatgabhane @dubielzyk-expensify @r8  when is the next review please?

File: docs/assets/js/main.js [hunk 8]
```diff
@@ -49,27 +49,41 @@ function isInRange(num, min, max) {
  * If a page is directly accessed (e.g., via deep link, bookmark, or opened in a new tab),the user will be navigated
  * back to the relevant hub page of that article.
  */
-function navigateBack() {
-    const currentHost = window.location.host;
-    const referrer = document.referrer;
```
Comment (rushatgabhane) [inline docs/assets/js/main.js:54]: @Taiwrash do you know why this code was removed?

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@Taiwrash bug 1: clicking any item in LHN takes you back 

https://github.com/user-attachments/assets/17ad7429-44ab-4e91-aa8d-afa684139831

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
Bug 2: if you click `back` from LHN, you can't go forward using browser forward button

**Comment by rushatgabhane** [comment]
@Taiwrash were you able to fix the bugs?

**Comment by rushatgabhane** [comment]
@shawnborton we can close this PR. This was completed by another contributor in another PR

**Comment by shawnborton** [comment]
Sounds good, will close.

**Pull request closed without merging**